superset/connectors/sqla/models.py

```diff
@@ -1465,6 +1465,12 @@ def has_extra_cache_key_calls(self, query_obj: QueryObjectDict) -> bool:
         for statement in templatable_statements:
             if ExtraCache.regex.search(statement):
                 return True
+
+        # Always recalculate the cache keys since the SQL query mutator can modify the
+        # query in arbitrary ways
+        if config["SQL_QUERY_MUTATOR"]:
+            return True
+
```

Comment:
This can have a very adverse effect on performance - rendering the query triggers all templated python code, which includes partition key checks etc. If your org is using the query mutator and is applying partition checks in the where clause, I fear this will seriously impact your DW. See our discussion from a while ago https://github.com/apache/superset/pull/7816#issuecomment-525403232 and a PR that fixed the problem https://github.com/apache/superset/pull/8128

Reply:
yeah, i think i'm going to abandon this PR, I agree the performance issues could be pretty bad